### PR TITLE
Repeating dataloader

### PR DESCRIPTION
**Pull request by le1nux**

### CODE REVIEW

File: src/modalities/dataloader/dataloader.py [hunk 1]
```diff
@@ -87,52 +79,56 @@ def fast_forward_batch_id(self) -> int:
...
-
-    @property
-    def fast_forward_sample_id(self) -> int:
-        """The sample id until which we fast-forward, as specified in the ResumableBatchSampler.
-
-        Returns:
-            int: fast forward sample id
-        """
-        return self.dataloader.batch_size * self.batch_sampler.start_index
+        return self.dataloader.batch_size
```
Comment (mali-git) [inline src/modalities/dataloader/dataloader.py:131]: ?

File: tests/dataloader/distributed/test_distributed_dataloader.py [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+        train_dataloader = components.train_dataloader
+        num_samples = len(train_dataloader.dataset)
+
+        batches = [batch.tolist() for batch in train_dataloader]
+
+        rank = dist.get_rank()
+        with open(f"tests/tmp/rank_{rank}_batches.json", "w") as f:
+            json.dump(batches, f)
+
+        with open("tests/tmp/rank_0_batches.json") as f:
```
Comment (mali-git) [inline tests/dataloader/distributed/test_distributed_dataloader.py:51]: Add lock

File: src/modalities/dataloader/dataloader.py (hunk 1 above)
Comment (le1nux) [reply src/modalities/dataloader/dataloader.py:131]: What was the confusion here?

File: tests/dataloader/distributed/test_distributed_dataloader.py (hunk 2 above)
Comment (le1nux) [reply tests/dataloader/distributed/test_distributed_dataloader.py:53]: done

File: tests/dataloader/distributed/test_distributed_dataloader.py [hunk 3]
```diff
@@ -0,0 +1,170 @@
...
+            rank_0_batches = torch.tensor(json.load(f))
+
+        with open("tests/tmp/rank_1_batches.json") as f:
+            rank_1_batches = torch.tensor(json.load(f))
+
+        samples = [i.item() for item in zip(rank_0_batches.flatten(), rank_1_batches.flatten()) for i in item]
+
+        assert len(rank_1_batches.flatten()) == num_samples // 2
+        assert len(set(samples)) == num_samples and num_samples > 0
+        assert set(samples).intersection(set(range(num_samples))) == set(range(num_samples))
```
[the remark this answers is not in the document]
Comment (le1nux) [reply tests/dataloader/distributed/test_distributed_dataloader.py:109]: done

**Review by mali-git — APPROVED** [review]

**Pull request merged**